From 25da4a6f5fd3566ae1b0bd5bc1a7bf645e169e6b Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Mon, 8 Nov 2010 15:27:04 +0800
Subject: [PATCH] ENGR00133485 mx28: low speed device doesn't be supported at otg port

Fix bug that low speed device doesn't be supported at otg port

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 arch/arm/plat-mxs/usb_common.c |    9 ++++-----
 1 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/arch/arm/plat-mxs/usb_common.c b/arch/arm/plat-mxs/usb_common.c
index d8da3dd..5e481d8 100644
--- a/arch/arm/plat-mxs/usb_common.c
+++ b/arch/arm/plat-mxs/usb_common.c
@@ -275,12 +275,11 @@ int usbotg_init(struct platform_device *pdev)
 	tmp &= ~PORTSC_PHCD;
 	__raw_writel(tmp, pdata->regs + UOG_PORTSC1);
 
-	if (pdata->operating_mode == FSL_USB2_DR_HOST) {
+	if ((pdata->operating_mode == FSL_USB2_DR_HOST) ||
+			(pdata->operating_mode == FSL_USB2_DR_OTG)) {
 		/* enable FS/LS device */
-		tmp = __raw_readl(IO_ADDRESS(pdata->phy_regs) + HW_USBPHY_CTRL);
-		tmp |= (BM_USBPHY_CTRL_ENUTMILEVEL2 |
-			BM_USBPHY_CTRL_ENUTMILEVEL3);
-		__raw_writel(tmp, IO_ADDRESS(pdata->phy_regs) + HW_USBPHY_CTRL);
+		__raw_writel(BM_USBPHY_CTRL_ENUTMILEVEL2 | BM_USBPHY_CTRL_ENUTMILEVEL3
+				, IO_ADDRESS(pdata->phy_regs) + HW_USBPHY_CTRL_SET);
 	}
 
 	otg_used++;
-- 
1.5.4.4

